### PR TITLE
Update the code for the bnn_hmc project

### PR DESCRIPTION
**Pull request by izmailovpavel**
We update the code for the bnn_hmc project.

### CODE REVIEW

**Comment by izmailovpavel** [comment]
@googlebot I signed it!

**Review by sharadmv — CHANGES_REQUESTED** [review]
Looks good, just a minor typo.

File: bnn_hmc/utils/losses.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  If num_classes is None, creates regression loss. Otherwise, creates
+  cross-entropy classification loss.
+  """
+  if num_classes:
+    return make_xent_log_likelihood(num_classes, temperature)
+  else:
+    return make_gaussian_likelihood(temperature)
+
+
+def make_xent_log_likelihood(temperature):
```
Comment (sharadmv) [inline bnn_hmc/utils/losses.py:37]: Should take in `num_classes`

File: bnn_hmc/utils/data_utils.py [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+"""Data loaders."""
+
+from typing import Generator, Tuple
+
+import jax
+import numpy as onp
+from jax import numpy as jnp
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from tensorflow.keras.datasets import imdb
```
Comment (sharadmv) [inline bnn_hmc/utils/data_utils.py:25]: Could you replace these tensorflow imports with the more internal-compatible:
```
imdb = tf.keras.datasets.imdb
sequence = tf.keras.preprocessing.sequence
```
Thanks!

File: bnn_hmc/make_posterior_surface_plot.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+  (likelihood_factory, predict_fn, ensemble_upd_fn, metrics_fns,
+   tabulate_metrics) = train_utils.get_task_specific_fns(task, data_info)
+  log_likelihood_fn = likelihood_factory(args.temperature)
+  log_prior_fn, log_prior_diff_fn = losses.make_gaussian_log_prior(
+      args.weight_decay, args.temperature)
+  
+  def eval(params, net_state, dataset):
+    likelihood, _ = log_likelihood_fn(
+      net_apply, params, net_state, dataset, is_training=True)
```
Comment (sharadmv) [inline bnn_hmc/make_posterior_surface_plot.py:110]: replace with just `True`

**Review by sharadmv — CHANGES_REQUESTED** [review]

File: bnn_hmc/utils/losses.py (hunk 1 above)
Comment (izmailovpavel) [reply bnn_hmc/utils/losses.py:37]: currently it infers `num_classes` from the shape of the logits, is there a reason to pass in `num_classes` as a parameter?
